kotlinx-coroutines-core/common/src/channels/Channel.kt

```diff
@@ -288,6 +289,15 @@ public interface ReceiveChannel<out E> {
      */
     public operator fun iterator(): ChannelIterator<E>
 
+    /**
+     * Emits all elements of this channel using [collector].
+     */
+    public suspend fun emitAll(collector: FlowCollector<E>) {
```

Comment:
new public method in public interface looks suspicious, maybe it is possible to avoid this?

Reply:
I agree. I'd say that we don't really want to diverge from the existing API to preserve compatibility as much as possible. We'd like to confine new API and keep it separate from the original library (see `IntellijCoroutines` object).